cmd/embedded-cluster/proxy.go

```diff
@@ -80,3 +93,81 @@ func setProxyEnv(proxy *ecv1beta1.ProxySpec) {
 		os.Setenv("NO_PROXY", proxy.NoProxy)
 	}
 }
+
+func includeLocalIPInNoProxy(c *cli.Context, proxy *ecv1beta1.ProxySpec) (*ecv1beta1.ProxySpec, error) {
+	if proxy != nil && (proxy.HTTPProxy != "" || proxy.HTTPSProxy != "") {
+		// if there is a proxy set, then there needs to be a no proxy set
+		// if it is not set, prompt with a default (the local IP or subnet)
+		// if it is set, we need to check that it covers the local IP
+		defaultIPNet, err := netutils.GetDefaultIPNet()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default IPNet: %w", err)
+		}
+		cleanDefaultIPNet, err := cleanCIDR(defaultIPNet)
+		if err != nil {
+			return nil, fmt.Errorf("failed to clean subnet: %w", err)
+		}
+		if proxy.ProvidedNoProxy == "" {
+			logrus.Infof("no-proxy was not set, adding the default interface's subnet %q to the no-proxy", cleanDefaultIPNet)
```

Comment:
```suggestion
			logrus.Infof("--no-proxy was not set. Adding the default interface's subnet (%q) to the no-proxy list.", cleanDefaultIPNet)
```

Reply:
Is it the default interface? I thought we said k0s doesn't necessarily use the default. We could say "the selected interface's subnet" or something similar.

Reply:
it's k0s's default interface \shrug